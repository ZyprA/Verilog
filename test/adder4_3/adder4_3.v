module adder4(a, b, s);
    input [3:0] a, b;
    output [3:0] s;

    assign s = a + b;

endmodule

module adder4_sim;
    reg [3:0] a, b;
    wire [3:0] s;

    adder4 a4(a, b, s);

    initial begin
        $dumpfile("adder4.vcd");
        $dumpvars(0, adder4_sim);
    end

    initial begin
        a = 4'b0000; b = 4'b0000;
        #100 a = 4'b0001;
        #100 a = 4'b0010;
        #100 b = 4'b0111;
        #100 a = 4'b1101;
        #100 a = 4'b1011;
        #100 $finish;
    end

endmodule