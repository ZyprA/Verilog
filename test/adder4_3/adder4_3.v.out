#! /opt/homebrew/Cellar/icarus-verilog/12.0/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/system.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_sys.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_textio.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/v2005_math.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/va_math.vpi";
S_0x125608300 .scope module, "adder4_sim" "adder4_sim" 2 9;
 .timescale 0 0;
v0x12561c5a0_0 .var "a", 3 0;
v0x12561c650_0 .var "b", 3 0;
v0x12561c700_0 .net "s", 3 0, L_0x12561c7d0;  1 drivers
S_0x125608470 .scope module, "a4" "adder4" 2 13, 2 1 0, S_0x125608300;
 .timescale 0 0;
    .port_info 0 /INPUT 4 "a";
    .port_info 1 /INPUT 4 "b";
    .port_info 2 /OUTPUT 4 "s";
v0x12560c910_0 .net "a", 3 0, v0x12561c5a0_0;  1 drivers
v0x12561c3e0_0 .net "b", 3 0, v0x12561c650_0;  1 drivers
v0x12561c490_0 .net "s", 3 0, L_0x12561c7d0;  alias, 1 drivers
L_0x12561c7d0 .arith/sum 4, v0x12561c5a0_0, v0x12561c650_0;
    .scope S_0x125608300;
T_0 ;
    %vpi_call 2 16 "$dumpfile", "adder4.vcd" {0 0 0};
    %vpi_call 2 17 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x125608300 {0 0 0};
    %end;
    .thread T_0;
    .scope S_0x125608300;
T_1 ;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0x12561c5a0_0, 0, 4;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0x12561c650_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0x12561c5a0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 2, 0, 4;
    %store/vec4 v0x12561c5a0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 7, 0, 4;
    %store/vec4 v0x12561c650_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 13, 0, 4;
    %store/vec4 v0x12561c5a0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 11, 0, 4;
    %store/vec4 v0x12561c5a0_0, 0, 4;
    %delay 100, 0;
    %vpi_call 2 27 "$finish" {0 0 0};
    %end;
    .thread T_1;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "adder4_3.v";
